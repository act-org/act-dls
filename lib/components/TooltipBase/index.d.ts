/**
 * @prettier
 */
import * as React from 'react';
import { TooltipProps } from '@material-ui/core';
export declare type Props = TooltipProps;
declare const TooltipBase: React.FC<Props>;
export default TooltipBase;
//# sourceMappingURL=index.d.ts.map