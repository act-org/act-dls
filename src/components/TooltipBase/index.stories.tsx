/**
 * @prettier
 */

import * as React from 'react';
import { action } from '@storybook/addon-actions';
import { boolean, select, text } from '@storybook/addon-knobs';

import TooltipBase from '.';

export const Base = (): React.ReactElement<any> => (
  <TooltipBase
    arrow={boolean('Has Arrow?', true)}
    onClose={action('onClose()')}
    onOpen={action('onOpen()')}
    open={boolean('Open?', true)}
    placement={select(
      'Placement',
      {
        Bottom: 'bottom',
        BottomEnd: 'bottom-end',
        BottomStart: 'bottom-start',
        Left: 'left',
        LeftEnd: 'left-end',
        LeftStart: 'left-start',
        Right: 'right',
        RightEnd: 'right-end',
        RightStart: 'right-start',
        Top: 'top',
        TopEnd: 'top-end',
        TopStart: 'top-start',
      },
      'top',
    )}
    title={text('Title', 'Tooltip title goes here.')}
  >
    <div>Tooltip Base</div>
  </TooltipBase>
);

export default {
  component: TooltipBase,
  parameters: {
    info: {
      text: 'This is the base variant of the Tooltip component.',
    },
  },
  title: 'Atoms/Tooltip',
};
