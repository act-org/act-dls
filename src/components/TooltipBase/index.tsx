/**
 * @prettier
 */

import * as React from 'react';
import { Tooltip, TooltipProps } from '@material-ui/core';

export type Props = TooltipProps;

const TooltipBase: React.FC<Props> = (
  props: Props,
): React.ReactElement<any> => <Tooltip {...props} />;

export default TooltipBase;
