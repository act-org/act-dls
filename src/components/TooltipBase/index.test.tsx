/**
 * @prettier
 */

import * as React from 'react';

import render from '~/helpers/test/render';

import TooltipBase from '.';

describe('TooltipBase', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <TooltipBase arrow placement="top" title="Tooltip title">
        <div>Tooltip Base</div>
      </TooltipBase>,
    );

    expect(container).toMatchSnapshot();
  });
});
