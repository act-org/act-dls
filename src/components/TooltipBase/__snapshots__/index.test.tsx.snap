// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TooltipBase matches the snapshot 1`] = `
<div>
  <div
    class=""
    title="Tooltip title"
  >
    Tooltip Base
  </div>
</div>
`;
